synapse/lib/snap.py

```diff
@@ -903,6 +907,118 @@ async def _raiseOnStrict(self, ctor, mesg, **info):
             raise ctor(mesg=mesg, **info)
         return False
 
+    async def _addNodeEdits(self, name, valu, props=None):
+
+        form = self.core.model.form(name)
+        if form is None:
+            raise s_exc.NoSuchForm(name=name)
+
+        if form.isrunt:
+            raise s_exc.IsRuntForm(mesg='Cannot make runt nodes.',
+                                   form=form.full, prop=valu)
+
+        if self.buidprefetch:
+            norm, info = form.type.norm(valu)
+            buid = s_common.buid((form.name, norm))
+            node = await self.getNodeByBuid(buid)
+            if node is not None:
+                if props is not None:
+                    return (node, await self.getNodeAdds(form, valu, props=props, addnode=False))
+                else:
+                    return (node, [(buid, form.name, [])])
+
+        return (None, await self.getNodeAdds(form, valu, props=props))
+
+    async def _addTagEdits(self, node, tags):
+
+        edits = []
+        for tag, valu in tags.items():
+
+            path = s_chop.tagpath(tag)
+            name = '.'.join(path)
+
+            if not await self.core.isTagValid(name):
+                await self.warn(f'Tag {tag} does not meet the regex for the tree.')
+                continue
+
+            tagnode = await self.addTagNode(name)
+
+            isnow = tagnode.get('isnow')
+            if isnow:
+                await self.warn(f'Tag {name} is now {isnow}')
+                name = isnow
+                path = isnow.split('.')
+
+            if isinstance(valu, list):
+                valu = tuple(valu)
+
+            if valu != (None, None):
+                try:
+                    valu = self.core.model.type('ival').norm(valu)[0]
+                except asyncio.CancelledError:  # pragma: no cover  TODO:  remove once >= py 3.8 only
+                    raise
+                except Exception as e:
+                    await self.warn(f'Bad tag value: #{tag}={valu}')
+                    continue
+
+            curv = None
+            if node is not None:
+                curv = node.tags.get(name)
+                if curv == valu:
+                    continue
```

Comment:
Should be able to mop up a few of these "bounce" coverage things by reloading the same data :)

Reply:
got the coverage bumped up to 100%